akka-serialization-jackson/src/main/scala/akka/serialization/jackson/JacksonSerializer.scala

```diff
@@ -445,30 +463,33 @@ import akka.util.Helpers.toRootLowerCase
   }
 
   def compress(bytes: Array[Byte]): Array[Byte] = {
-    compressionAlgorithm match {
-      case Compression.Off => bytes
-      case Compression.GZip(largerThan) =>
-        if (bytes.length > largerThan) compressGzip(bytes) else bytes
+    @inline def shouldCompress: Boolean = compressionAlgorithm match {
+      case Compression.Off              => false
+      case Compression.GZip(largerThan) => bytes.length > largerThan
+      case Compression.LZ4(largerThan)  => bytes.length > largerThan
     }
-  }
-
-  private def compressGzip(bytes: Array[Byte]): Array[Byte] = {
+    if (!shouldCompress) return bytes
     val bos = new ByteArrayOutputStream(BufferSize)
-    val zip = new GZIPOutputStream(bos)
+    val zip = compressionAlgorithm match {
+      case Compression.GZip(_) => new GZIPOutputStream(bos)
+      case _                   => new LZ4FrameOutputStream(bos)
+    }
     try zip.write(bytes)
     finally zip.close()
     bos.toByteArray
   }
 
   def decompress(bytes: Array[Byte]): Array[Byte] = {
-    if (isGZipped(bytes))
-      decompressGzip(bytes)
-    else
-      bytes
-  }
+    if (!isGZipped(bytes) && !isLZ4(bytes)) {
```

Comment:
duplicated test as below.

Reply:
Did you mean I also tested if the `bytes` are compressed in
```
    val in = if (isGZipped(bytes)) {
      new GZIPInputStream(bais)
    } else {
      new LZ4FrameInputStream(bais)
    }
```

The test in `(!isGZipped(bytes) && !isLZ4(bytes)) ` is for early return if the `bytes` are not compressed at all, while the test below is to check whether the `bytes` are compressed with `gzip` or with `lz4`. I can check all the conditions in one `if` statement, it is unclear to me which style is better.